draft-ietf-ppm-dap.md

```diff
@@ -1612,37 +1670,55 @@ This structure includes the following:
 
 * The number of reports included in the batch.
 
-* The vector of encrypted aggregate shares. They MUST appear in the same order
-  as the aggregator endpoints list of the task parameters.
+* The vector of Leader and Helper encrypted aggregate shares. They MUST appear
+  in the same order as the aggregator endpoints list of the task parameters.
+
+If obtaining Helper aggregate shares fails, then the Leader responds to
+subsequent POST requests to the collection with an HTTP error status and a
+problem document as described in {{errors}}.
+
+The Leader MAY respond with 204 No Content for requests to a collection if the
+results have been deleted due to age.
+
+###### A note on idempotence
+
+The reason we use a POST instead of a GET to poll the state of a collection is
+because of the fixed-size query mode (see {{fixed-size-query}}). Collectors may
+make a query against the current batch, and it is the Leader's responsibility to
+keep track of what batch is current for some task. Polling a collection is the
+only point at which it is safe for the Leader to change its current batch, since
+it constitutes acknowledgement on the Collector's part that it received the
+response to some previous PUT request to the collections resource.
```

Comment:
I don't think I follow. For starters, you're talking about polling the collect URI here, correct? If yes, then the contents of the response, that is the aggregate result, is not going to change once computed.

Reply:
Yes, but that's not what affects idempotence. What matters is that polling a collection has the _side effect_ of changing what the leader considers to be the current batch. There's a much more verbose explanation of this [here](https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/blob/134d5305849f52a59061a543e843387564b24a69/api-resources.md?plain=1#L341).